File: data/readable_format/cpp/file1152.cpp
#include <iostream>
#include <vector>

// Declaration and definition of the peak function
void peak(std::vector<float>& peaksu, float peaki, int iel, std::vector<int>& ieltmp, int& icnt) {
    if (peaki > 1.0e-05) {
        icnt++;
        peaksu[icnt - 1] = peaki; // Adjusted for 0-based indexing
        ieltmp[icnt - 1] = iel;   // Adjusted for 0-based indexing
    }
}

int main() {
    std::vector<float> peaksu(3);
    float peaki = 0.00002; // Example peak value
    int iel = 2; // Example element index
    std::vector<int> ieltmp(3);
    int icnt = 0; // Counter for peaks

    // Call the peak function
    peak(peaksu, peaki, iel, ieltmp, icnt);

    // Output the results
    std::cout << "icnt: " << icnt << std::endl;
    std::cout << "peaksu: ";
    for (int i = 0; i < icnt; ++i) {
        std::cout << peaksu[i] << " ";
    }
    std::cout << std::endl << "ieltmp: ";
    for (int i = 0; i < icnt; ++i) {
        std::cout << ieltmp[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1152.f90
program test_peak
  implicit none
  real :: peaki = 0.00002
  real, dimension(3) :: peaksu
  integer :: iel = 2
  integer, dimension(3) :: ieltmp
  integer :: icnt = 0
  integer :: i

  ! Initialize arrays to zero
  peaksu = 0.0
  ieltmp = 0

  call peak(peaksu, peaki, iel, ieltmp, icnt)

  print *, "icnt:", icnt
  print *, "peaksu:"
  do i = 1, icnt
    print *, peaksu(i)
  end do
  print *, "ieltmp:"
  do i = 1, icnt
    print *, ieltmp(i)
  end do
end program test_peak

subroutine peak(peaksu,peaki,iel,ieltmp,icnt)
  implicit none
  real, intent(in) :: peaki
  real, dimension(3), intent(inout) :: peaksu
  integer, intent(in) :: iel
  integer, dimension(3), intent(inout) :: ieltmp
  integer, intent(inout) :: icnt

  if (peaki.gt.1.0e-05) then
    icnt = icnt + 1
    peaksu(icnt) = peaki
    ieltmp(icnt) = iel
  end if

  return
end subroutine peak